src/mediastreambroker/DeviceControllerBasedMediaStreamBroker.ts

```diff
@@ -2,10 +2,16 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import DeviceController from '../devicecontroller/DeviceController';
+import EventController from '../eventcontroller/EventController';
 import MediaStreamBroker from '../mediastreambroker/MediaStreamBroker';
 
 export default interface DeviceControllerBasedMediaStreamBroker
   extends DeviceController,
     MediaStreamBroker {
   // This interface combines a device controller with a media stream broker
+
+  /**
+   * EventController for publishing events.
+   */
+  eventController?: EventController | undefined;
```

Comment:
Why does this have to be in the interface level?

Reply:
Otherwise if a builder implements their own `DeviceController` that has an `EventController` we won't be able to use it unless it's defined in the interface

Reply:
Do we need access to it outside of the device controller class?

Reply:
We use it in order to keep the EventController between the MeetingSession and DeviceController the same. https://github.com/aws/amazon-chime-sdk-js/blob/acd66ec4783114c0452162b77d8dab479e1c2f18/src/meetingsession/DefaultMeetingSession.ts#L52

Reply:
Should it be in the DeviceController interface then?

Reply:
Yes we can move it to the DeviceController interface